frontend/src/bundles/common/components/toggle/toggle.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { useFormController } from '~/bundles/common/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+    control: Control<T, null>;
+    label: string;
+    name: FieldPath<T>;
+    type?: 'checkbox' | 'radio' | 'switch';
+};
+
+const Toggle = <T extends FieldValues>({
+    control,
+    label,
+    name,
+    type = 'checkbox',
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({ name, control });
```

Comment:
The component should not know anything about the environment in which it is used

Reply:
Now components are purely cosmetic